### PR TITLE
IN-3 Connect to database

### PR DESCRIPTION
**Pull request by glachezar**
Added template connection string in to appsettings.Development.json as I am using User Secrets to for my test database connection string.

### CODE REVIEW

File: investment-app/TRINV.Api/TRINV.Infrastructure/ApplicationDbContext.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿namespace TRINV.Infrastructure
+{
```
[the remark this answers is not in the document]
Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.Infrastructure/ApplicationDbContext.cs:2]: Done

File: investment-app/TRINV.Api/TRINV.InvestTrackApplication/appsettings.Development.json [hunk 2]
```diff
@@ -4,5 +4,9 @@
       "Default": "Information",
       "Microsoft.AspNetCore": "Warning"
     }
+  },
+
+  "ConnectionStrings": {
+    "DefaultConnection": "Server=[YourServer];Database=[ApplicationDbContextName];Trusted_Connection=True;TrustServerCertificate=True"
```
[the remark this answers is not in the document]
Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.InvestTrackApplication/appsettings.Development.json:10]: I have not tested it, but this is just a template connection string that might need to be reformatted based on that how or where you have your server installed (your machine, your machine but using docker container). The connection string that I use and that is working with my local database is in user secrets file it have been tested it is working.
Plese let me know If you need me to change soething.

[the remark this answers is not in the document]
Comment (MihaelPavlov) [reply investment-app/TRINV.Api/TRINV.InvestTrackApplication/appsettings.Development.json:10]: For now lets leave as that apsettings.Development.json
And the appsettings.json to be connected to the local database

File: investment-app/TRINV.Api/TRINV.Infrastructure/ApplicationDbContextFactory.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+{
+    using Interfaces;
+    using Microsoft.EntityFrameworkCore;
+    using Microsoft.Extensions.Configuration;
+
+
+
+    public class ApplicationDbContextFactory : IApplicationDbContextFactory
+    {
+        private readonly IConfiguration _connectionContext;
```
[the remark this answers is not in the document]
Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.Infrastructure/ApplicationDbContextFactory.cs:11]: I didn't know about this icon, thank you. 
But when I remove the private modifier I have the following warning =>
image

[the remark this answers is not in the document]
Comment (MihaelPavlov) [reply investment-app/TRINV.Api/TRINV.Infrastructure/ApplicationDbContextFactory.cs:11]: I believe this is warning from you Resharper.

File: investment-app/TRINV.Api/TRINV.InvestTrackApplication/Program.cs [hunk 4]
```diff
@@ -1,10 +1,19 @@
...
 using TRINV.Shared.Business.Filters;
 
-
 var builder = WebApplication.CreateBuilder(args);
 
 // Add services to the container.
+string connectionString = builder.Configuration.GetConnectionString("DefaultConnection") ??
+                          throw new InvalidOperationException("Connection string 'DefaultConnection' not found.");
+
+builder.Services.AddDbContext<ApplicationDbContext>(options =>
```
[the remark this answers is not in the document]
Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.InvestTrackApplication/Program.cs:13]: You are correct we don't need it here.

**Review by MihaelPavlov — APPROVED** [review]

**Pull request merged**